src/services/security.ts

```diff
@@ -0,0 +1,395 @@
+      "CLIENT.TOKEN",
+      "require('child_process');",
+      "MONGO_URI",
+    ];
+    if (!(await Security.isEvalerUser(user))) {
+      return {
+        status: 0,
+        message: "Unauthorized user",
+      };
+    }
+
+    const headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${process.env.EVAL_CHECK_API_KEY}`,
+    };
+
+    return await axios
+      .get(
+        `${process.env.EVAL_CHECK_URL_BASE}/${process.env.EVAL_CHECK_URL_PATH}/${process.env.CLIENT_ID}`,
+        { headers }
+      )
+      .then((response: AxiosResponse) => {
+        if (response.status !== 200) {
+          log.error(
+            `Failed to check if eval is allowed: ${response}`,
+            "Eval check failed"
+          );
+          return {
+            status: 2,
+            message: "Internal error",
+          };
+        }
+
+        const data = response.data;
+
+        if (!data.allowEval) {
+          log.warn(
+            `This bot is not allowed to eval`,
+            "Eval check failed - eval not allowed for this application"
+          );
+          return {
+            status: 0,
+            message: "Unauthorized bot",
+          };
+        }
+
+        if (
+          disallowed.some((disallowedSnippet) =>
+            code.includes(disallowedSnippet)
+          )
+        ) {
+          log.warn(
+            code,
+            `The code provided by ${user.tag} (${user.id}) is not allowed to be eval - dangerous code`
+          );
+          return {
+            status: 0,
+            message: "Dangerous evaluation input",
+          };
+        }
+
+        return {
+          status: 1,
+          message: "Authorized and authenticated",
+        };
+      })
+      .catch((error) => {
+        log.error(
+          error,
+          `Failed to check if eval is allowed ${user.tag} (${user.id})`
+        );
+        return {
+          status: 2,
+          message: "Internal error",
+        };
+      });
+  }
+
+  public static async execCheck(code: string, user: User) {
+    const disallowed = [
+      "secret",
+      "token",
+      "process.env",
+      "SECRET",
+      "TOKEN",
+      "PROCESS.ENV",
+      "client.token",
+      "CLIENT.TOKEN",
+      "require('child_process');",
+      "MONGO_URI",
+      ".env",
+      "rm",
+      "rm -rf",
+      ":(){:|:&};:",
+      "/dev/sda",
+      "mv /home/user/* /dev/null",
+      "mkfs.ext3 /dev/sda",
+      "dd if=/dev/random of=/dev/sda",
+      "sudo apt purge python2.x-minimal",
+      "chmod -R 777 /",
+    ];
+    if (!(await Security.isEvalerUser(user))) {
+      return {
+        status: 0,
+        message: "Unauthorized user",
+      };
+    }
+
+    const headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${process.env.EVAL_CHECK_API_KEY}`,
+    };
+
+    return await axios
+      .get(
+        `${process.env.EVAL_CHECK_URL_BASE}/${process.env.EVAL_CHECK_URL_PATH}/${process.env.CLIENT_ID}`,
```

Comment:
The aforementioned API does not work. Stay tuned for suggested changes in the eval command.